src/main/java/org/dependencytrack/parser/trivy/TrivyParser.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * This file is part of Dependency-Track.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) Steve Springett. All Rights Reserved.
+ */
+package org.dependencytrack.parser.trivy;
+
+import alpine.common.logging.Logger;
+import org.dependencytrack.model.Cwe;
+import org.dependencytrack.model.Severity;
+import org.dependencytrack.model.Vulnerability;
+import org.dependencytrack.model.VulnerableSoftware;
+import org.dependencytrack.parser.common.resolver.CweResolver;
+import org.dependencytrack.parser.trivy.model.Bitnami;
+import org.dependencytrack.persistence.QueryManager;
+
+import java.math.BigDecimal;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+public class TrivyParser {
+
+    private static final Logger LOGGER = Logger.getLogger(TrivyParser.class);
+
+    public Vulnerability parse(org.dependencytrack.parser.trivy.model.Vulnerability data, QueryManager qm) {
+        Vulnerability synchronizedVulnerability = new Vulnerability();
+        Vulnerability vulnerability = new Vulnerability();
+        List<VulnerableSoftware> vsList = new ArrayList<>();
+
+        Vulnerability.Source source = Vulnerability.Source.NVD;
+
+        if (data.getVulnerabilityID().startsWith("GHSA-")) {
+            source = Vulnerability.Source.GITHUB;
+        }
+
+        vulnerability.setSource(source);
+
+        vulnerability.setPatchedVersions(data.getFixedVersion());
+
+        // get the id of the data record (vulnerability)
+        vulnerability.setVulnId(data.getVulnerabilityID());
+        vulnerability.setTitle(data.getTitle());
+        vulnerability.setDescription(data.getDescription());
+        vulnerability.setSeverity(parseSeverity(data.getSeverity()));
+
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.ENGLISH);
+
+        if (data.getPublishedDate() != null) {
+            try {
+                vulnerability.setPublished(formatter.parse(data.getPublishedDate()));
+            } catch (ParseException ex) {
+                LOGGER.error("Unable to parse published date %s".formatted(data.getPublishedDate()));
```

Comment:
Is this an ERROR level problem? It'd say at most WARNING.

Reply:
thanks for the feedback, fixed